### ISSUE TITLE
[fresh_graphql] wrong auth status

### ISSUE THREAD

**Issue by subzero911** [felangel/fresh#82]
At the first time I do setToken(), status always turns to "unauthorized".
After refresh it becomes authorized.

code:
```dart
final freshLink = FreshLink.oAuth2(
    tokenStorage: InMemoryTokenStorage(),
    refreshToken: (token, client) async {
      // Perform refresh and return new token
      log.d('refreshing token!');
      return OAuth2Token(accessToken: accessToken, tokenType: 'Bearer'); // TODO refresh with RefreshToken
    },
    shouldRefresh: (response) => true, // TODO implement
  )..authenticationStatus.listen(log.d);
  await freshLink.setToken(OAuth2Token(accessToken: accessToken, tokenType: 'Bearer'));
```

I debugged it and cannot find an issue, but the problem is definitely in `setToken()`.
I await it and suddenly execution goes to `void _updateStatus` where `token` is null !

**Comment by subzero911** [comment]
3) That's extremely strange, but if I subscribe to status AFTER setToken(), it works just fine:

<img width="924" alt="image" src="https://user-images.githubusercontent.com/12999702/229218778-a3c87bcf-3b53-44e0-8e90-2870aa7669ef.png">
Result:
<img width="611" alt="image" src="https://user-images.githubusercontent.com/12999702/229218890-eb285d2f-9453-4fff-b58b-8b34db9390b1.png">

Maybe it's because of this line
<img width="530" alt="image" src="https://user-images.githubusercontent.com/12999702/229219885-609e74d4-c8df-4105-a016-0e6f202f03be.png">
When I create a FreshLink, the setter is run, and  it suddenly turns status to "unauthenticated". And for some reason it does not change afterwards.
Just guessing.

**Comment by felangel** [comment]
This has been fixed in #96
